### PR TITLE
"ARIA required context role" and "ARIA required owned elements" (ff89c9 / bc4a75): Update note on DPUB due to upcoming changes in 1.1

### PR DESCRIPTION
**Pull request by Jym77**
The inapplicable example about ARIA DPUB role was failing the requirement because the `doc-biblioentry` role was not owned by its required `doc-bibliography`. This resulted as implementation considering the ARIA DPUB module (such as Alfa) to be rejected as they were failing this example.

Closes issue(s):

- N/A

Need for Final Call:
This will require a 1 week Final Call (updating an example)

---

### **When creating PR:**

- [X] Make sure you're requesting to **pull a branch** (right side) to the `develop` branch (left side).

### **After creating PR:**

- [X] Add yourself (and co-authors) as "Assignees" for PR.
- [X] Add label to indicate if it's a `Rule`, `Definition` or `Chore`.
- [ ] [Link the PR to any issue it solves](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue). This will be done automatically by [referencing the issue](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) at the top of this comment in the indicated place.
- [X] Optionally request feedback from anyone in particular by assigning them as "Reviewers".

### **When merging a PR:**

- [ ] Close any issue that the PR resolves. This will happen automatically upon merging if the PR was correctly linked to the issue, e.g. by referencing the issue at the top of this comment.

## How to Review And Approve

- Go to the “Files changed” tab
- Here you will have the option to leave comments on different lines.
- Once the review is completed, find the “Review changes” button in the top right, select “Approve” (if you are really confident in the rule) or "Request changes" and click “Submit review”.
- Make sure to also review the proposed Final Call period. In case of disagreement, the longer period wins.

### CODE REVIEW

**Review by kasperisager — APPROVED** [review]

**Review by carlosapaduarte — APPROVED** [review]

File: _rules/aria-required-context-role-ff89c9.md
Comment (WilcoFiers) [inline _rules/aria-required-context-role-ff89c9.md:216]: This should not be changed. The point of this example is to demonstrate that "owned by" as used in ARIA means any descendant, not just the child. This example comes directly from the DPUB ARIA spec:

https://www.w3.org/TR/dpub-aria-1.0/#doc-biblioentry

This confusion about "owned by" including all descendants, and not just the children and direct references of aria-owns is a known issue in ARIA. See: w3c/aria#1161, w3c/aria#748.

**Review by WilcoFiers — CHANGES_REQUESTED** [review]

File: _rules/aria-required-context-role-ff89c9.md
Comment (Jym77) [reply _rules/aria-required-context-role-ff89c9.md:216]: 🙈 That example has so many problems I want to unsee it…

Fortunately (maybe), `doc-biblioentry` is deprecated in DPUB-ARIA 1.1 (https://w3c.github.io/dpub-aria/#doc-biblioentry) precisely because of the wonkyness it creates with the `list` role (https://github.com/w3c/dpub-aria/issues/15 and https://github.com/w3c/dpub-aria/pull/22). This is very recent changes!

I guess this example should be completely revamped. As far as I can tell, there is no more required context roles in DPUB-ARIA 1.1. I don't think it is a good idea to keep an example showing a known problem in the specs which is already fixed in the next version.

If the point of the example is to show the "descendant" meaning of "owned by", we need to find a non DPUB example.

If the point of the example is to show that DPUB roles are ignored, we can take any, but they have no required context role…

Maybe we should update the Background note saying that DPUB-ARIA 1.1 does not have any required context role (like Graphics ARIA) which justifies our choice to restrict ourselves to ARIA 1.1 roles.

**Comment by Jym77** [comment]
I've updated the background explaining why we focus on ARIA. Now saying that DPUB 1.1 has no role with required context role. Copied that note to "required owned element" rule where the same thing happen. Essentially, the "unresolved issues" that this note mentioned have been resolved by deprecating the offending elements.

The example has been updated accordingly.

Dismissing all reviews, since this is a significant change of scope.

**Review by kasperisager — APPROVED** [review]

File: _rules/aria-required-context-role-ff89c9.md
Comment (WilcoFiers) [inline _rules/aria-required-context-role-ff89c9.md:46]: DPUB-ARIA 1.1 hasn't even made it to first draft yet. I don't think we should be referencing it.

File: _rules/aria-required-context-role-ff89c9.md
Comment (WilcoFiers) [inline _rules/aria-required-context-role-ff89c9.md:213]: I disagree with changing this. The point of this example is to show a DPUB role that has a required context that is inapplicable.

**Review by WilcoFiers — DISMISSED** [review]
I think this is getting ahead of things, and I don't think the new example is a proper replacement of the current one.

File: _rules/aria-required-context-role-ff89c9.md
Comment (Jym77) [reply _rules/aria-required-context-role-ff89c9.md:213]: But in that example, the `<div role="list">` owns no element with a role of `listitem`. Because role inherintance doesn't work like that and ARIA DPUB 1.0 is broken on that point, and this is known, and this is the reason we advance for excluding DPUB roles. See also https://github.com/w3c/dpub-aria/issues/15 for more details.

So, we have an Inapplicable example that does not satisfy the Accessibility Requirement (1.3.1), which goes directly against our test cases design guidance… https://github.com/act-rules/act-rules.github.io/blob/develop/pages/design/test-cases.md
> This requires that **passed and inapplicable test cases should satisfy the accessibility requirements** of the rule

(emphasise is not mine)

So that example should be killed anyway.

Comment (WilcoFiers) [reply _rules/aria-required-context-role-ff89c9.md:213]: Aaahhh. I see what you mean. Sorry, took me a while. Yeah that's a fair point. That `list` is invalid here.

**Review by WilcoFiers — APPROVED** [review]

**Review by carlosapaduarte — APPROVED** [review]

**Comment by Jym77** [comment]
Final Call ends on November 4.

**Comment by Jym77** [comment]
Final Call has ended. Merging.

**Pull request merged**
